diffrax/_solver/langevin_srk.py

```diff
@@ -0,0 +1,325 @@
+    _LangevinDriftTerm,
+    _LangevinTuple,
+    _LangevinX,
+    AbstractTerm,
+    MultiTerm,
+    WrapTerm,
+)
+from .base import AbstractItoSolver, AbstractStratonovichSolver
+
+
+_ErrorEstimate = TypeVar("_ErrorEstimate", None, _LangevinTuple)
+_LangevinArgs = tuple[_LangevinX, _LangevinX, Callable[[_LangevinX], _LangevinX]]
+
+
+def get_args_from_terms(
+    terms: MultiTerm[tuple[AbstractTerm[Any, RealScalarLike], AbstractTerm]],
+) -> _LangevinArgs:
+    drift, diffusion = terms.terms
+    if isinstance(drift, WrapTerm):
+        unwrapped_drift = drift.term
+        assert isinstance(diffusion, WrapTerm)
+        unwrapped_diffusion = diffusion.term
+        assert isinstance(unwrapped_drift, _LangevinDriftTerm)
+        assert isinstance(unwrapped_diffusion, _LangevinDiffusionTerm)
+        gamma = unwrapped_drift.gamma
+        u = unwrapped_drift.u
+        f = unwrapped_drift.grad_f
+    else:
+        assert isinstance(drift, _LangevinDriftTerm)
+        assert isinstance(diffusion, _LangevinDiffusionTerm)
+        gamma = drift.gamma
+        u = drift.u
+        f = drift.grad_f
+    return gamma, u, f
+
+
+class AbstractCoeffs(eqx.Module):
+    dtype: AbstractVar[jnp.dtype]
+
+
+_Coeffs = TypeVar("_Coeffs", bound=AbstractCoeffs)
+
+
+# TODO: I'm not sure if I can use the _Coeffs type here,
+# given that I do not use Generic[_Coeffs] in the class definition.
+# How should I work around this?
+class SolverState(eqx.Module, Generic[_Coeffs]):
+    h: RealScalarLike
+    taylor_coeffs: PyTree[_Coeffs, "_LangevinX"]
+    coeffs: _Coeffs
+    rho: _LangevinX
+    prev_f: _LangevinX
+
+
+# CONCERNING COEFFICIENTS:
+# The coefficients used in a step of this SRK depend on
+# the time increment h, and the parameter gamma.
+# Assuming the modelled SDE stays the same (i.e. gamma is fixed),
+# then these coefficients must be recomputed each time h changes.
+# Furthermore, for very small h, directly computing the coefficients
+# via the function below can cause large floating point errors.
+# Hence, we pre-compute the Taylor expansion of the SRK coefficients
+# around h=0. Then we can compute the SRK coefficients either via
+# the Taylor expansion, or via direct computation.
+# In short the Taylor coefficients give a Taylor expansion with which
+# one can compute the SRK coefficients more precisely for a small h.
+
+
+class AbstractLangevinSRK(
+    AbstractStratonovichSolver[SolverState],
+    AbstractItoSolver[SolverState],
+    Generic[_Coeffs, _ErrorEstimate],
+):
+    """Abstract class for Stochastic Runge Kutta methods specifically designed
+    for Underdamped Langevin Diffusion of the form
+
+    $$d x_t = v_t dt$$
+
+    $$d v_t = - gamma v_t dt - u ∇f(x_t) dt + (2gammau)^(1/2) dW_t$$
+
+    where $v$ is the velocity, $f$ is the potential, $gamma$ and $u$ are the
+    friction and momentum parameters, and $W$ is a Brownian motion.
+
+    Solvers which inherit from this class include ALIGN, SORT, ShOULD, and
+    QUIC_SORT.
```

Comment:
Can you turn these into links in the documentation?

This can be done by styling them as e.g. ``[`diffrax.ALIGN`][]``